### PR TITLE
Set gas price in gwei

### LINKED ISSUE #1381: Edit gas price in GWEI
I think we knew this, but I got a nice kick in the pants from [this article](https://medium.com/@ethgasstation/the-safe-low-gas-price-fb44fdc85b91), which should be required reading for any MetaMask dev, about how we should think about handling gas price input.

### PR DESCRIPTION
**Pull request by danfinlay**
Also enforces "safe low gas" minimum recommended by this article by eth-gas-station:
https://medium.com/@ethgasstation/the-safe-low-gas-price-fb44fdc85b91

Fixes #1381

Also now returns state string from `logState()` enabling `copy(logState())` to copy state to clipboard.

Fix interation warning from React in ens component.

### CODE REVIEW

File: ui/app/components/pending-tx.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ const addressSummary = util.addressSummary
 const nameForAddress = require('../../lib/contract-namer')
 const HexInput = require('./hex-as-decimal-input')
 
-const MIN_GAS_PRICE_BN = new BN(20000000)
+const MIN_GAS_PRICE_GWEI_BN = new BN(2)
+const GWEI_FACTOR = new BN(Math.pow(10, 9))
```
Comment (kumavis) [inline ui/app/components/pending-tx.js]: did you know that you can write `Math.pow(10, 9)` as `1e9`? neato eh

Comment (flyswatter) [inline ui/app/components/pending-tx.js]: Oh man that's so great how did I take so long to learn that?

File: ui/app/components/pending-tx.js [hunk 2]
```diff
@@ -39,16 +41,20 @@ PendingTx.prototype.render = function () {
...
+  // Gas
   const gas = txParams.gas
-  const gasPrice = txParams.gasPrice
-
   const gasBn = hexToBn(gas)
+
+  // Gas Price
+  const gasPrice = txParams.gasPrice || MIN_GAS_PRICE_BN.toString(16)
   const gasPriceBn = hexToBn(gasPrice)
+  const gasPriceGweiBn = gasPriceBn.div(new BN(Math.pow(10, 9)))
```
Comment (kumavis) [inline ui/app/components/pending-tx.js]: maybe reuse `GWEI_FACTOR` here

Comment (flyswatter) [inline ui/app/components/pending-tx.js]: Both comments received, implemented & pushed.

File: ui/app/components/pending-tx.js [hunk 3]
```diff
@@ -187,17 +193,18 @@ PendingTx.prototype.render = function () {
               }, [
                 h(HexInput, {
                   name: 'Gas Price',
-                  value: gasPrice,
-                  suffix: 'WEI',
-                  min: MIN_GAS_PRICE_BN.toString(10),
+                  value: gasPriceGweiBn.toString(16),
+                  suffix: 'GWEI',
+                  min: MIN_GAS_PRICE_GWEI_BN.toString(10),
```
Comment (kumavis) [inline ui/app/components/pending-tx.js]: are we correctly mixing decimals and hex here? (`value` and `min`)

**Review by kumavis — APPROVED** [review]

**Comment by Zanibas** [comment]
Adjusting the price is not impacting what is being changed in the transaction being sent. May be related to #1314

**Review by 2-am-zzz — CHANGES_REQUESTED** [review]
Gas params are not being changed in the final tx.

**Review by 2-am-zzz — APPROVED** [review]

**Pull request merged**